src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -1060,6 +1060,12 @@ object SymDenotations {
       else defn.RootClass
     }
 
+    /** Is this denotation public ? */
+    def isPublic(implicit ctx: Context): Boolean =
+      // private-qualified denotations do not have the Private flag set,
+      // see `ParserCommon#normalize`
+      !this.is(Protected | Private) && !privateWithin.exists
```

Comment:
This link behaves weirdly so copy-pasting:


Because we remove Private when a qualifier exists: https://github.com/lampepfl/dotty/blob/master/src/dotty/tools/dotc/parsing/Parsers.scala#L1422-L1427 scalac does the same thing. I guess this is because private qualified are not private in the bytecode, maybe we should do this in a miniphase later.